src/Compiler/Reports/AbstractReportManager.cpp

```diff
@@ -106,16 +104,18 @@ int AbstractReportManager::parseErrorWarningSection(QTextStream &in, int lineNr,
 
   QString line;
   // Store line numbers in set to always know which one was first
-  std::set<int> wrnLines, errLines;
+  std::map<int, QString> warnings, errors;
```

Comment:
Are you sure you didn't reorder the elements by using map/set? As we discussed we should keel the order of elements.

Reply:
Yes. both std::set and std::map use std::less<Key> comparator by default.
So we end up with the earliest message/error going first in the container.

Reply:
Recent commit implements following grouping:
![image](https://user-images.githubusercontent.com/109239890/207139607-9de53bf5-2f23-4c68-bf43-8c7d2575784d.png)

Errs/warnings, that aren't split by any Info message, are grouped within a single item. For standalone warnings/errors no grouping is made.